assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.test.js

```diff
@@ -121,6 +128,116 @@ describe( 'WidgetAreaRenderer', () => {
 		} );
 	} );
 
+	it( 'should only render widgets the user has access to in a view-only viewContext', async () => {
+		createWidgets( registry, areaName, [
+			{
+				Component: WidgetComponent,
+				modules: 'search-console',
+				slug: 'one',
+				width: WIDGET_WIDTHS.FULL,
+			},
+			{
+				Component: WidgetComponent,
+				modules: 'search-console',
+				slug: 'two',
+				width: WIDGET_WIDTHS.FULL,
+			},
+			{
+				Component: () => <div>AdSense is here</div>,
+				modules: 'adsense',
+				slug: 'three',
+				width: WIDGET_WIDTHS.FULL,
+			},
+		] );
+
+		// Add our dashboard sharing capabilities into state.
+		fetchMock.getOnce(
+			/^\/google-site-kit\/v1\/core\/modules\/data\/list/,
+			{
+				body: FIXTURES,
+				status: 200,
+			}
+		);
+		const capabilitiesBaseVar = '_googlesitekitUserData';
+		const capabilitiesWithPermission = {
+			permissions: {
+				googlesitekit_view_dashboard: true,
+				googlesitekit_manage_options: true,
+				'googlesitekit_read_shared_module_data::["adsense"]': false,
+				'googlesitekit_read_shared_module_data::["search-console"]': true,
+			},
+		};
+		global[ capabilitiesBaseVar ] = capabilitiesWithPermission;
```

Comment:
Similarly, let's use `provideUserCapabilities` here instead of the global.

Reply:
Ah, interesting, there's another test using this so maybe I'll update that one at the same time if possible 👍🏻